subsys/bluetooth/controller/ll_sw/ull_sched.c

```diff
@@ -571,19 +638,40 @@ static uint8_t after_match_slot_get(uint8_t user_id, uint32_t ticks_slot_abs,
 {
 	uint32_t ticks_to_expire_prev;
 	uint32_t ticks_slot_abs_prev;
+	uint32_t ticks_anchor_prev;
 	uint32_t ticks_to_expire;
 	uint8_t ticker_id_prev;
 	uint8_t ticker_id;
+	uint8_t retry;
 
+	/* As we may place the event between two other events, ensure there is
+	 * space for 3 times +/- 16 us jitter. I.e. with 32KHz sleep clock,
+	 * ~30.517 us after previous event, ~30.517 us before and after current
+	 * event, and an ~30.517 us before next event. Hence 8 time of ceil
+	 * value 16 us (30.517 / 2).
+	 */
 	ticks_slot_abs += HAL_TICKER_US_TO_TICKS(EVENT_JITTER_US << 3);
 
+	/* There is a possibility that ticker nodes expire during iterations in
+	 * this function causing the reference ticks_anchor returned for the
+	 * found ticker to change. In this case the iterations have to be
+	 * restarted with the new reference ticks_anchor value.
+	 * Simultaneous continuous scanning on 1M and Coded PHY, alongwith
+	 * directed advertising and one other state/role could expire in quick
+	 * succession, hence have a retry count of 4.
```

Comment:
I am still not sure why we are using a retry count of 4, compared to e.g. 3 or 5 or some other number. This comment only states that we need a retry, but not really how many.

Reply:
As mentioned in the comment, retry 4 times due to ticker expiry with 1M, Coded PHY, directed advertising and one other state/role (can be central/peripheral/advertiser/etc.). Preemption of the iterations (while loop) by ISR from these ticker expiry need the loop to restart to find the ticker offsets all over.

Reply:
Ah, that makes sense. Was not aware of that detail :)